inference-engine/src/cldnn_engine/ops/tensor_iterator.cpp

```diff
@@ -0,0 +1,200 @@
+    }
+}
+
+void CreateTensorIteratorOp(Program &p, const std::shared_ptr<TensorIterator> &op) {
+    // loop can takes multiple inputs, no p.ValidateInputs()
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+
+    // set body topology
+    InferenceEngine::CNNNetwork body_network(op->get_body());
+    Program body_program(body_network, p.GetEnginePtr(), p.GetConfig(), false);
+    auto body_topology = *body_program.GetTopology();
+
+    // setup input_mappings/ output_mappings and back_edges
+    const auto& loop_input_descs = op->get_input_descriptions();
+    const auto& loop_output_descs = op->get_output_descriptions();
+    const auto& body_inputs = op->get_body()->get_parameters();
+    const auto& body_outputs = op->get_body()->get_results();
+
+    std::vector<cldnn::loop::primitive_mapping> input_mappings;
+    std::vector<cldnn::loop::primitive_mapping> output_mappings;
+    std::vector<cldnn::loop::backedge_mapping> back_edges;
+
+    // set input mapping & back edges
+    for (const auto& loop_input_desc : loop_input_descs) {
+        const cldnn::primitive_id& external_id = inputPrimitives.at(loop_input_desc->m_input_index);
+        auto& body_input = body_inputs.at(loop_input_desc->m_body_parameter_index);
+        cldnn::primitive_id internal_id = layer_type_name_ID(body_input);
+
+        // set input mapping
+        if (const auto& sliceInfo =
+            std::dynamic_pointer_cast<TensorIterator::SliceInputDescription>(loop_input_desc)) {
+            // sliced input
+            input_mappings.emplace_back(external_id, internal_id, sliceInfo->m_axis,
+                sliceInfo->m_start, sliceInfo->m_end, sliceInfo->m_stride);
+        } else {
+            // InvariantInputDescription or InputDescription
+            // input without slicing
+            input_mappings.emplace_back(external_id, internal_id);
+        }
+
+        // set back edges
+        if (const auto& mergedInput =
+            std::dynamic_pointer_cast<TensorIterator::MergedInputDescription>(loop_input_desc)) {
+            // backedge
+            const auto& to = body_inputs.at(mergedInput->m_body_parameter_index);
+            const auto& from = body_outputs.at(mergedInput->m_body_value_index);
+            cldnn::primitive_id to_id = layer_type_name_ID(to);
+            cldnn::primitive_id from_id = layer_type_name_ID(from);
+            back_edges.emplace_back(from_id, to_id);
+        }
+    }
+
+    // set trip count, initial execution condition, num iteration primitives
+    // they should be mutable_data to prevent from being optimized out
+    std::string layerName = layer_type_name_ID(op);
+    const cldnn::primitive_id trip_count_id = layerName + "_tripCount";
+    const int64_t num_iterations = op->get_num_iterations();
+    assert(num_iterations >= 0);
+    {
+        cldnn::mutable_data trip_count = CreateIntData(p, trip_count_id, num_iterations);
+        p.primitivesToIRLayersMap[trip_count_id] = { op->get_friendly_name() };
```

Comment:
Seems that trip_count is not needed to be mutable

Reply:
updated trip_count to be data instead of mutable_data